### PR TITLE
Add animations to step indicator

### PR DESCRIPTION
**Pull request by peso3**
Also changed so it no longer uses a constraint layout

### CODE REVIEW

File: warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt [hunk 1]
```diff
@@ -1,281 +1,580 @@
...
-                    top.linkTo(parent.top)
-                }
-            }
-            Box(
-                modifier = Modifier.constrainAs(title) {
-                    top.linkTo(dot.top)
-                    bottom.linkTo(dot.bottom)
-                    start.linkTo(dot.end)
+    require(steps > 1)
+    val colors = WarpTheme.colors.components.stepIndicator
```
Comment (julbar) [inline warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt:67]: I think we should use the CompositionLocalProvider to get the colors

File: warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt [hunk 2]
```diff
@@ -1,281 +1,580 @@
...
+    val animatedStep = animateFloatAsState(targetValue = activeStep.toFloat(), label = "lineAnimation").value
+    val dimensions = WarpTheme.dimensions.components.stepIndicator
```
Comment (julbar) [inline warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt:69]: I think we should use the CompositionLocalProvider to get the dimensions

File: warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt [hunk 3]
```diff
@@ -1,281 +1,580 @@
...
+                                }
+
+                            }
+                        }
+                        // Indicator index is greater than the one we are animating towards
+                        (i > floor(animatedStep.toDouble() + 1)) -> {
+                            drawCircle(
+                                colors.background,
+                                style = Stroke(dimensions.placeholderBorderWidth.dp.toPx()),
+                                radius = size.minDimension/2f // - dimensions.placeholderBorderWidth.dp.toPx()/2
```
Comment (julbar) [inline warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt:190]: Forgotten code?

File: warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt [hunk 4]
```diff
@@ -1,281 +1,580 @@
...
+                            )
+                        }
+                        // Indicator is the one we are currently animating towards
+                        else -> {
+                            val frac = animatedStep - (i - 1)
+                            // First draw a regular indicator
+                            drawCircle(
+                                color = colors.background,
+                                style = Stroke(dimensions.placeholderBorderWidth.dp.toPx()),
+                                radius = size.minDimension/2f // - dimensions.placeholderBorderWidth.dp.toPx()/2
```
Comment (julbar) [inline warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt:200]: Forgotten code?

File: warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt [hunk 5]
```diff
@@ -1,281 +1,580 @@
...
 @Composable
-fun Dot(
+fun HorizontalWarpStepIndicator(
     modifier: Modifier = Modifier,
-    state: StepState,
-    onClick: (() -> Unit)?
+    steps: Int,
+    activeStep: Int = 0,
+    onStepClicked: ((Int) -> Unit)? = null,
+    label: @Composable ((Int) -> Unit)? = null
```
Comment (julbar) [inline warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt:399]: I think this should just be a string so the the styling of the label is in our control

File: warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt [hunk 6]
```diff
@@ -1,281 +1,580 @@
...
-        modifier = (if (onClick != null) modifier
-            .clip(CircleShape)
-            .clickable { onClick() } else modifier)
-    ) {
-        when (state) {
-            StepState.Active -> {
-                Canvas(modifier = Modifier.size(24.dp)) {
-                    drawCircle(completedColor)
+    require(steps > 1)
+    val colors = WarpTheme.colors.components.stepIndicator
```
Comment (julbar) [inline warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt:402]: I think we should use the CompositionLocalProvider to get the colors

File: warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt [hunk 7]
```diff
@@ -1,281 +1,580 @@
...
+    val animatedStep = animateFloatAsState(targetValue = activeStep.toFloat(), label = "lineAnimation").value
+    val dimensions = WarpTheme.dimensions.components.stepIndicator
```
Comment (julbar) [inline warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt:404]: I think we should use the CompositionLocalProvider to get the dimensions

**Review by julbar — CHANGES_REQUESTED** [review]

File: warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt [hunk 8]
```diff
@@ -1,281 +1,653 @@
...
+ * @param steps Amount of steps. Must be greater than 1
  * @param activeStep Currently active step
- * @param orientation Orientation to draw the steps in
  * @param onStepClicked will be invoked if a step is clicked
- * @param stepTitle Optional composable that will be drawn with the step
- * @param stepDescription Optional composable that will be drawn below the title
+ * @param stepTitle Function to provide titles for the steps
+ * @param stepContent Composable containing the content of each step
  */
 @Composable
```
Comment (julbar) [inline warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt:65]: We need to support xml aswell, so we need to add a AbstractComposeView class of this component

**Review by julbar — CHANGES_REQUESTED** [review]

File: warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt (hunk 8 above)
Comment (peso3) [reply warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt:65]: Yeah I know, this PR only adds animations though so I would say it is out of scope for this PR

**Review by julbar — APPROVED** [review]

**Pull request merged**